misc/python/materialize/cli/mzcompose.py

```diff
@@ -625,7 +626,16 @@ def handle_composition(
             # test analytics, even if the workflow doesn't define more granular
             # test cases.
             with composition.test_case(f"workflow-{args.workflow}"):
-                composition.workflow(args.workflow, *args.unknown_subargs[1:])
+                ci_extra_args = json.loads(os.getenv("CI_EXTRA_ARGS", "{}"))
+                buildkite_step_key = os.getenv("BUILDKITE_STEP_KEY")
+                extra_args = (
+                    ci_extra_args[buildkite_step_key].split(" ")
```

Comment:
Why do you split at the space? Should it be the colon?

Reply:
Seems more natural to list the arguments the same way as you'd do on command line.

Reply:
Ah, got it. This is the part we are applying it to: ` "--scenario=Foasdb"`

Reply:
Actually shlex.split is even better: https://docs.python.org/3/library/shlex.html#shlex.split will use that.